qiskit_nature/properties/second_quantization/electronic/electronic_driver_result.py

```diff
@@ -0,0 +1,103 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The ElectronicDriverResult class."""
+
+from typing import List, Tuple, cast
+
+from qiskit_nature.drivers import Molecule
+from qiskit_nature.drivers.second_quantization import QMolecule
+from qiskit_nature.operators.second_quantization import FermionicOp
+
+from ..driver_result import DriverResult
+from ..second_quantized_property import LegacyDriverResult, LegacyElectronicDriverResult
+from .angular_momentum import AngularMomentum
+from .bases import ElectronicBasis, ElectronicBasisTransform
+from .dipole_moment import TotalDipoleMoment
+from .electronic_energy import ElectronicEnergy
+from .magnetization import Magnetization
+from .particle_number import ParticleNumber
+
+
+class ElectronicDriverResult(DriverResult):
+    """The ElectronicDriverResult class.
+
+    This is a :class:~qiskit_nature.properties.CompositeProperty gathering all property objects
+    previously stored in Qiskit Nature's `QMolecule` object.
+    """
+
+    def __init__(self) -> None:
+        """
+        Property objects should be added via `add_property` rather than via the initializer.
+        """
+        super().__init__(self.__class__.__name__)
+        self.molecule: Molecule = None
+        self.electronic_basis_transform: ElectronicBasisTransform = None
+        # TODO: add origin driver metadata
```

Comment:
This was done originally for the persisted HDF5 file so there was information in there as to its origin inside the file itself in case you wanted to say re-run the original driver or whatever. I do not know what you have in mind around any persistence here in regards of properties - if anything.

Reply:
I see, this makes sense. As outlined in the description of this PR I want to (re-)introduce the ability to store/load HDF5 files as I think this is an important feature to have. I agree that it is important to have this information somewhere as it provides context where the data came from.
Do you think a simple dictionary like the one above is sufficient?

Reply:
@woodsp-ibm pinging about:
> Do you think a simple dictionary like the one above is sufficient?

Reply:
The properties were all string types before where they could contain multi-line strings like the config for PSI4 or G16. For PySCF & PyQuante the config was more a string formatted like a kwargs since its a programmatic interface. Arguably the driver name, version and config should be enough info to be able to recreate the data that was there in the HDF5 as it represented with what and how it was created in the first place - it was really all about having it identified in the HDF5 and available when it was read back so you knew the orginal origin of data if you wanted to know. The dictionary you present captures all that was there before - so in that regard it seems to suffice.